web/src/left_sidebar_navigation_area.ts

```diff
@@ -130,6 +130,12 @@ export function handle_narrow_activated(filter: Filter): void {
 function toggle_condensed_navigation_area(): void {
     const $views_label_container = $("#views-label-container");
     const $views_label_icon = $("#toggle-top-left-navigation-area-icon");
+
+    if (page_params.is_spectator) {
+        // If the user is a spectator, do nothing
+        return;
+    }
```

Comment:
Is this code necessary? I would have thought it would be unreachable with the button being hidden...

Reply:
Yes, I think it is necessary. I have removed the hidden-for-spectator class from the div with id "views-label-container" so that my leftbar-side-title ("Views") would be visible to spectators and further to prevent the toggling of the view-label-container it was necessary to add the above written code. Since the previously written code was already handling what needs to be hidden from the spectators (like inbox, mentions, etc.) in the expanded view the removal of hidden-for-spectator from views-label-container shouldn't be a problem and toggling wont be possible due to the added code in the ts file.

Reply:
@timabbott can you please review my approach and provide me your views on it?

Reply:
OK, this does seem correct to not have clicking "VIEWS" do something, but we should also disable `cursor: pointer` on that element for spectators.

As a side note, I think you could have made to product impact a lot more clear in your answer. Let's make the comment say something explaining the why "// We don't support collapsing VIEWS for spectators, so exit early."